### PR TITLE
refactor: remove code duplication in expiration date

### PR DESCRIPTION
**Pull request by dawidgil**

### CODE REVIEW

File: src/app/shared/expiration/date-picker/expiration-date.component.ts [hunk 1]
```diff
@@ -78,15 +78,12 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
...
     this.add.emit(this.defaultValidityPeriod);
-    this.expirationDate.setValue(
-      new Date(Date.now() + this.defaultValidityPeriod)
-    );
-    this.expirationTimeShift = this.defaultValidityPeriod;
     this.hideRemoveButton = false;
   }
 
-  defaultExpirationDate() {
+  defaultExpirationDate(): void {
```
Comment (jrhender) [inline src/app/shared/expiration/date-picker/expiration-date.component.ts]: I'm curious what your personal convention is for using the `set` prefix? Why (or why not) should this have `set` prefix (i.e. be `setDefaultExpirationDate()`). It seems as though `this.expirationDate` is being set so maybe it makes sense?

File: src/app/shared/expiration/date-picker/expiration-date.component.ts [hunk 2]
```diff
@@ -78,15 +78,12 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
   }
 
   restoreDefaultHandler(): void {
+    this.defaultExpirationDate();
     this.add.emit(this.defaultValidityPeriod);
-    this.expirationDate.setValue(
-      new Date(Date.now() + this.defaultValidityPeriod)
-    );
-    this.expirationTimeShift = this.defaultValidityPeriod;
```
Comment (jrhender) [inline src/app/shared/expiration/date-picker/expiration-date.component.ts:85]: This line doesn't seem to be in the new logic. Is this expected?

File: src/app/shared/expiration/date-picker/expiration-date.component.ts [hunk 3]
```diff
@@ -78,15 +78,12 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
   }
 
   restoreDefaultHandler(): void {
+    this.defaultExpirationDate();
     this.add.emit(this.defaultValidityPeriod);
```
Comment (jrhender) [inline src/app/shared/expiration/date-picker/expiration-date.component.ts]: This is a bit odd to me. `expirationDate` is being set but we're emitting `defaultValidityPeriod`. But maybe I'm missing some pattern/convention here. Does it make sense to you?

**Review by jrhender — COMMENTED** [review]
Thanks for trying out the refactor! Left a few questions

File: src/app/shared/expiration/date-picker/expiration-date.component.ts (hunk 3 above)
Comment (dawidgil) [reply src/app/shared/expiration/date-picker/expiration-date.component.ts]: Yeah, defaultExpirationDate was setting expirationDate to be equal default, so it worked, but logically it wasn't correct. Changed it by adding method:
```
  private emitExpirationTimeShift() {
    this.add.emit(this.expirationTimeShift);
  }
```

File: src/app/shared/expiration/date-picker/expiration-date.component.ts (hunk 2 above)
Comment (dawidgil) [reply src/app/shared/expiration/date-picker/expiration-date.component.ts:85]: It is, in method `setDefaultExpirationDate` (changed it from `defaultExpirationDate`).

File: src/app/shared/expiration/date-picker/expiration-date.component.ts (hunk 1 above)
Comment (dawidgil) [reply src/app/shared/expiration/date-picker/expiration-date.component.ts]: added prefix, you're right. I'm setting there a value, so it's better to add `set`.

**Review by jrhender — APPROVED** [review]
Looks good, thanks!

**Pull request merged**